platform/persistence/platform-persistence-core-listeners/src/main/java/org/codice/ddf/persistence/events/AlertListener.java

```diff
@@ -0,0 +1,189 @@
+        this.persistentStore = persistentStore;
+    }
+
+    public void init() throws UnknownHostException {
+        this.hostName = InetAddress.getLocalHost()
+                .getHostName();
+    }
+
+    /**
+     * Processes alert events and stores them in the persistent store. Because this handler 'squashes'
+     * alerts into one entry it needs to be synchronized to be thread safe.
+     *
+     * @param event The osgi alert event
+     * @throws IllegalArgumentException
+     */
+    @Override
+    public synchronized void handleEvent(Event event) throws IllegalArgumentException {
+        if (Alert.ALERT_DISMISS_TOPIC.equals(event.getTopic())) {
+            dismissAlert(event);
+            return;
+        }
+        //need source to lookup previous alerts
+        if (event.getProperty(Alert.SYSTEM_NOTICE_SOURCE_KEY) == null) {
+            return;
+        }
+        LOGGER.debug("Received alert event on topic {}", event.getTopic());
+
+        Alert alert = getAlertFromEvent(event);
+        Alert existingAlert = getAlertBySource(alert.getSource());
+        if (existingAlert != null) {
+            alert.setId(existingAlert.getId());
+            alert.setCount(existingAlert.getCount() + 1);
+            alert.setTime(existingAlert.getTime());
+        }
+
+        addAlertToStore(alert);
+    }
+
+    private void addAlertToStore(Alert alert) {
+        PersistentItem item = new PersistentItem();
+        item.addIdProperty(alert.getId());
+        item.addProperty(Alert.SYSTEM_NOTICE_SOURCE_KEY, alert.getSource());
+        item.addProperty(Alert.SYSTEM_NOTICE_HOST_NAME_KEY, alert.getHostName());
+        item.addProperty(Alert.SYSTEM_NOTICE_HOST_ADDRESS_KEY, alert.getHostAddress());
+        item.addProperty(Alert.SYSTEM_NOTICE_PRIORITY_KEY,
+                alert.getPriority()
+                        .value());
+        item.addProperty(Alert.SYSTEM_NOTICE_TITLE_KEY, alert.getTitle());
+        item.addProperty(Alert.SYSTEM_NOTICE_DETAILS_KEY, alert.getDetails());
+        item.addProperty(Alert.SYSTEM_NOTICE_TIME_KEY, alert.getTime());
+        item.addProperty(Alert.ALERT_STATUS, alert.getStatus());
+        item.addProperty(Alert.ALERT_LAST_UPDATED, alert.getLastUpdated());
+        item.addProperty(Alert.ALERT_COUNT, alert.getCount());
+
+        if (alert.getDismissedBy() != null) {
+            item.addProperty(Alert.ALERT_DISMISSED_BY, alert.getDismissedBy());
+            item.addProperty(Alert.ALERT_DISMISSED_TIME, alert.getDismissedTime());
+        }
+
+        try {
+            persistentStore.add(CORE_NAME, item);
+        } catch (PersistenceException e) {
+            LOGGER.error("Failed to persist alert.");
+        }
+    }
+
+    private Alert getAlertFromEvent(Event event) {
+        Map<String, Object> properties = new HashMap<>();
+        for (String name : event.getPropertyNames()) {
+            properties.put(name, event.getProperty(name));
+        }
+        return new Alert(properties);
```

Comment:
❓ Should we consider passing the `Event` to the `Alert` and let it extract what it needs?